plugins/package-managers/node/src/funTest/assets/projects/synthetic/npm-babel-expected-output.yml

```diff
@@ -2552,13 +2552,13 @@ packages:
       algorithm: "SHA-1"
   vcs:
     type: "Git"
-    url: "https://github.com/babel/babel/tree/master/packages/babel-cli"
+    url: "https://github.com/babel/babel.git"
```

Comment:
Commit message:
- "Improve *the* algorithm"
- "by deriving the path either from the given URL in addition to from" -> "by deriving the path from the URL in addition to"

Reply:
Looking purely at the data without having looked at the code changes yet, this does not look like an improvement to me, because:
- The previous (unprocessed) `vcs` is now processed, i.e. not taken as-is from metadata anymore. Even if the URL was wrong in the original metadata, it should be taken here, as it's the sole responsibility of `vcs_processed` to "correct" the metadata.
- We loose information about what branch was referred in the original metadata.

Reply:
> this does not look like an improvement to me, because:

@mnonnenmacher  do you also see all aspects of this PR as "not an improvement" / "undersirable", or do you see certain aspects of it which make sense? e.g. we've talked mostly about the revision so far, but also `vcs.path` and `vcs.type` do change, as well as the URLs.

Reply:
> do you also see all aspects of this PR as "not an improvement" / "undersirable"

For the record, I'm not seeing *all* aspects of this PR as a whole negatively. The first commit is a nice refactoring, and as [noted](https://github.com/oss-review-toolkit/ort/pull/8333#discussion_r1500430367) used the URL as a fallback for determining the path also makes sense to me.

Reply:
I've split out that first commit to https://github.com/oss-review-toolkit/ort/pull/8336.

Reply:
> Looking purely at the data without having looked at the code changes yet, this does not look like an improvement to me, because:
> 
>     * The previous (unprocessed) `vcs` is now processed, i.e. not taken as-is from metadata anymore. Even if the URL was wrong in the original metadata, it should be taken here, as it's the sole responsibility of `vcs_processed` to "correct" the metadata.
> 
>     * We loose information about what branch was referred in the original metadata.

I tried to start working on this PR again, but I'm having difficulties solving the issue you mention here. 
Let me try to explain. I understand your ask as follows:

The `Package.vcs` field should only hold the data exactly as specified in package metadata.

That ask implies to me that the function extracting the VCS info `NpmSupport: parseNpmVcsInfo()` must not process it, but processing must happen separately. The `parseNpmVcsInfo()` is a couple of callers, but looking just at the caller inside `Npm` already shows that your ask is already not the case, because the algorithm:

1. Gathers package metadata in `var vcsFromPackage`
2. Gathers VCS from download in `val vcsFromDownloadUrl`
3. Gathers VCS from file system directory in `val vcsFromDirectory`
4. merges 2 or 3 into 1.
5. Uses the result als the (unprocessed) `Package.vcs`

So, what you are asking from my code is already before heavily violation. 
I'm not objecting your ask, but wanted to double check whether you are aware that
this is basically asking for a bit larger refactoring. I can do that, but I would like to agree
upfront, what exactly has to go into `Package.vcs`. (My guess is that only things from package metadata should go there. ) 

Are we that strict as well in other package managers? (I would also be open for the approach to not refactor it and leave the change basically as-is) What do you think?

Reply:
> The `Package.vcs` field should only hold the data exactly as specified in package metadata.

Yes, at least that's how the property is currently documented:

https://github.com/oss-review-toolkit/ort/blob/d39d163d136ddd9d8661f5c1350f2a3644ccde16/model/src/main/kotlin/Package.kt#L109-L112

Note, though, that at least to me "package metadata" does not exclude that more sources (like a package registry) than just `package.json` could be consulted for metadata.

But it's a valid question where to draw the line to `Package.vcsProcessed` then.

> I would like to agree upfront, what exactly has to go into Package.vcs.

Makes sense. Could you please add it to the agenda of the next Kotlin dev meeting?